src/org/zowe/jenkins_shared_library/pipelines/generic/GenericPipeline.groovy

```diff
@@ -1431,6 +1433,19 @@ class GenericPipeline extends Pipeline {
         //     preSetupException = new PackagingStageException("arguments.localWorkspace is not defined for packagingGeneric", arguments.name)
         // }
 
+        Integer causeID = Pipeline.build.getCause()
+        this.isAuthorizedUser = false
+        if (causeID == PipelineConstants.USERID_CAUSE_ID) {
+            this.isAuthorizedUser = true
+        } else if (causeID == PipelineConstants.BRANCHEVENT_CAUSE_ID || causeID == PipelineConstants.BRANCHINDEXING_CAUSE_ID) {
+            if (this.changeInfo.isPullRequest) {
+                this.isAuthorizedUser = isPRAuthorizedUser()
+            }
+        }
+        if (!this.isAuthorizedUser) {
+            preSetupException = new PackagingStageException("this.isAuthorizedUser is not true, user not authorized to run build", this.isAuthorizedUser)
```

Comment:
We can say automatic packaging step for non-committers on z/OS is disabled.

Reply:
ok